tests/test_threads.py

```diff
@@ -180,3 +181,16 @@ def test_should_abort():
 
         should_abort()
         ThreadPool().query_abort.called_once_with()
+
+
+def test_stop_threads():
+    with patch("guikit.threads.ThreadPool", MagicMock()):
+        from guikit.threads import ThreadPool
+
+        _COUNT = 3
+        for _ in range(_COUNT):
+            ThreadPool().run_thread(lambda: threading.sleep(0.5))
+
+        assert len(ThreadPool()._workers) == _COUNT
+        ThreadPool().stop_threads()
+        assert not any(worker.is_alive for worker in ThreadPool()._workers)
```

Comment:
This is a unit test so, in reality, you don't want to actually create a bunch of threads and check if they stop: what you want is to check that all workers have the `abort` property set to `True` and that their `join` method has been called.